src/tests/test_fastod.cpp

```diff
@@ -0,0 +1,624 @@
+#include <algorithm>
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/od/fastod/canonical_od.h"
+#include "algorithms/od/fastod/fastod.h"
+#include "all_csv_configs.h"
+#include "config/names.h"
+#include "config/time_limit/type.h"
+#include "csv_config_util.h"
+
+namespace tests {
+
+inline size_t combine_hashes(size_t first, size_t second) {
+    size_t wave = second + 2654435769UL + (first << 6) + (first >> 2);
+    return first ^ wave;
+}
+
+size_t RunFastod(CSVConfig const& csv_config) {
+    config::InputTable input_table = MakeInputTable(csv_config);
+
+    algos::fastod::DataFrame data = algos::fastod::DataFrame::FromInputTable(input_table);
+    algos::fastod::Fastod fastod(std::move(data));
```

Comment:
Why we avoid algo factory here? 
And in general, it’s interesting why we can do the same thing in two ways - move a dataframe from the outside (pass to the constructor), or create it inside the algorithm from an input table.
It seems to me that we need to leave only the second way. In other algorithms, we usually create algorithm-specific data within the algorithm based on the input table, and then work with this data. The only exception that I know of is pli-based fd algorithms. We can load data using the `LoadData` method (see `class PliBasedFDAlgorithm`). But I'm not sure if it's a similar case here.

Reply:
Removed the ability to pass DataFrame to Fastod constructor. Now we use algorithm factory to create algorithm.